indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,546 @@
+                            'class': 'just-use-option'
+                        })
+                        .on('click', function() {
+                            node.trigger('typeahead:click-on-custom-option', {item_name: node.val()});
+                        });
+                        resultsWrapper.append(justUse);
+                    }
+                    justUse.text($T.gettext('Just use "{0}"').format(node.val()));
+                } else {
+                    justUse.remove();
+                }
+
+                /* Always mark the first option as active in order for the keyboard inputs to work */
+                var firstOption = resultHtmlList.find('li.typeahead-item').first();
+                if (!firstOption.length) {
+                    node.parent().find('.just-use-option').addClass('active');
+                    justUseOptionSelected = true;
+                } else {
+                    resultHtmlList.find('li.typeahead-item').first().addClass('active');
+                    justUseOptionSelected = false;
+                }
+                return resultHtmlList;
+            }
+
+            function scrollResultsList(inputField) {
+                var parent = inputField.parent();
+                var activeOption = parent.find('.typeahead-results').find('.typeahead-item.active');
+                var dropdownList = parent.find('.typeahead-list');
+                var resultsListHeight = parent.find('.typeahead-results').outerHeight();
+                if (activeOption.length != 0) {
+                    if (activeOption.position().top + activeOption.outerHeight() > dropdownList.outerHeight()) {
+                        delta = activeOption.position().top + activeOption.outerHeight() - dropdownList.outerHeight();
+                        currentScrollPos += delta;
+                        dropdownList.animate({
+                            scrollTop: currentScrollPos
+                        }, 50);
+                    } else if (activeOption.position().top < dropdownList.position().top){
+                        currentScrollPos += activeOption.position().top;
+                        dropdownList.animate({
+                            scrollTop: currentScrollPos
+                        }, 50);
+                    }
+                }
+                parent.find('.just-use-option').toggleClass('active', !activeOption.length);
+                justUseOptionSelected = !activeOption.length;
+            }
+
+            function postSelectionActions(inputField, value) {
+                field.val(JSON.stringify(hiddenData)).trigger('change');
+                if (inputField.is(venueInput)) {
+                    latestVenueValue = value;
+                } else if (inputField.is(roomInput)) {
+                    latestRoomValue = value;
+                }
+            }
+
+            function highlightOption(inputField, value) {
+                var highlighter = inputField.siblings('.keyword-highlighter');
```

Comment:
it's a nice idea and it totally works. For more readability I will keep the current code, but if there's a strong opinion, I can change it